unlock-app/src/components/interface/locks/Settings/forms/UpdateHooksForm.tsx

```diff
@@ -74,6 +88,52 @@ const HookMapping: Record<FormPropsKey, HookValueProps> = {
   },
 }
 
+interface HookSelectProps {
+  label: string
+  name: string
+  disabled: boolean
+}
+
+const HookSelect = ({ name, label, disabled }: HookSelectProps) => {
+  const [selectedOption, setSelectedOption] = useState<string | null>(null)
+
+  return (
+    <ConnectForm>
+      {({ register, getValues, formState: { errors } }: any) => {
+        const value = getValues(name)
+        const showInput =
+          (value?.length > 0 && value !== ZERO) ||
+          (selectedOption ?? '')?.length > 0
+
```

Comment:
Let's actually move that logic _out_ of this component.

Think about the next steps: we will have more UI: one for each of the contract (password hook, discount... etc) so let's make sure our code is ready for this.

Here we need to create a `CustomContractHook` component that is displayed only when the input value is `CUSTOM_ADDRESS`.